src/os/filestore/FileStore.cc

```diff
@@ -3199,9 +3201,14 @@ int FileStore::_do_fiemap(int fd, uint64_t offset, size_t len,
       extent->fe_length = offset + len - extent->fe_logical;
     (*m)[extent->fe_logical] = extent->fe_length;
     i++;
-    extent++;
+    last = extent++;
   }
   free(fiemap);
+  if (!(last->fe_flags & FIEMAP_EXTENT_LAST)) {
+    offset = last->fe_logical + last->fe_length;
+    len -= offset;
```

Comment:
This len -= line looks wrong... it assumes the original offset was 0, which is isn't (esp if you end up doing 2 batches of fiemaps).